src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -213,11 +213,25 @@ export class AccountManagementService implements IAccountManagementService {
 
 			let result = await self._accountStore.addOrUpdate(account);
 			if (result.accountAdded) {
+				// Double check that there isn't a matching account
+				if (result.changedAccount!.key.authLibrary === MSAL_AUTH_LIBRARY) {
```

Comment:
Since this is only for MSAL, changing back to ADAL keeps the old accounts as-is.. with email comparison we can get rid of this check too.